src/main/java/net/openhft/chronicle/queue/internal/util/InternalFileUtil.java

```diff
@@ -254,4 +268,25 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
             return FileVisitResult.CONTINUE;
         }
     }
+
+    private static class ProcFdWalkerWithPid extends ProcFdWalker {
```

Comment:
would it be simpler to add a method to ProcFdWalker to do something with the file e? and then this class can just implement the action method instead of overriding the whole method?

Reply:
Good point - ta!

If the parent class is deprecated it'll make the actioning the deprecation a little more complex, but cleaner in the meantime.